benches/benches/bevy_ecs/components/change_detection.rs

```diff
@@ -0,0 +1,219 @@
+use bevy_ecs::{
+    component::Component,
+    entity::Entity,
+    prelude::{Added, Changed},
+    world::World,
+};
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use rand::{prelude::SliceRandom, SeedableRng};
+use rand_chacha::ChaCha8Rng;
+
+#[derive(Component, Default)]
+#[component(storage = "Table")]
+struct Table;
```

Comment:
These are ZSTs. Even with black_box, not sure if the compiler won't take some shortcuts. Could we add a f32 inside these and just increment them for mutation?

Reply:
Thanks, I did your suggested change ; I had to make a new trait to allow the modification ; and I realized I had a mistake on my generic type usage